addons/point_of_sale/models/pos_order.py

```diff
@@ -905,15 +909,30 @@ def create_from_ui(self, orders, draft=False):
         :type draft: bool.
         :Returns: list -- list of db-ids for the created and updated orders.
         """
-        order_ids = []
+        orders_extra_data = {}
         for order in orders:
             existing_order = False
             if 'server_id' in order['data']:
+                ## Suggestion of replacement:
+                ## existing_order = search(id = order['data']['server_id'], limit=1)
+                ## if not existing_order:
+                ##    existing_order = search(pos_reference = order['data']['name'], limit=2)
+                ## if len(existing_order) != 1:
+                ##    existing_order = False
+                ## This way, we identify in priority an order by it's id, otherwise we use the only order that has the same pos_reference, assuming that it's the same order, otherwise we don't take the risk to pick the wrong order and we create a new one with the same pos_reference.
```

Comment:
Yes. You can apply this fix/refactoring.

Reply:
Since the create_from_ui implementation has changed with a different behavior, I removed my suggestion.